shiva/src/main/scala/nl/lumc/sasc/biopet/pipelines/shiva/variantcallers/somatic/MuTect2.scala

```diff
@@ -0,0 +1,101 @@
+import nl.lumc.sasc.biopet.extensions.gatk.SelectVariants
+import nl.lumc.sasc.biopet.extensions.igvtools.IGVToolsIndex
+import nl.lumc.sasc.biopet.utils.IoUtils
+import nl.lumc.sasc.biopet.utils.config.Configurable
+import org.broadinstitute.gatk.queue.extensions.gatk.TaggedFile
+
+class MuTect2(val parent: Configurable) extends SomaticVariantcaller {
+
+  def name = "mutect2"
+
+  override val mergeVcfResults: Boolean = false
+
+  // currently not relevant, at the moment only one somatic variant caller exists in Biopet
+  // and results from this won't be merged together with the results from other methods
+  def defaultPrio = -1
+
+  //val buildPONFromNormals: Boolean = config("build_PON_from_normals", default = false)
+
+  val runConEst = config("run_contest", default = "false")
+
+  def biopetScript(): Unit = {
+    val samplesDir: File = new File(outputDir, "samples")
+    if (!samplesDir.exists()) samplesDir.mkdir()
+
+    var renameSamples: List[String] = List()
+    var tumorSamples: List[String] = List()
+    var intermResult: File = null
+
+    if (tnPairs.size == 1) {
+      val pair = tnPairs.head
+      renameSamples = List(s"TUMOR ${pair.tumorSample}")
+      tumorSamples = List("TUMOR")
+      intermResult = new File(samplesDir, s"${pair.tumorSample}-${pair.normalSample}.$name.vcf")
+      addJobForPair(pair, intermResult)
+
+    } else {
+      var outputPerSample: List[TaggedFile] = List()
+      for (pair <- tnPairs) {
+        val pairLabel = s"${pair.tumorSample}-${pair.normalSample}"
+        val out: File = new File(samplesDir, s"$pairLabel.$name.vcf")
+        renameSamples :+= s"TUMOR.$pairLabel ${pair.tumorSample}"
+        tumorSamples :+= s"TUMOR.$pairLabel"
+        outputPerSample :+= TaggedFile(out, pairLabel)
+        addJobForPair(pair, out)
+      }
+
+      var sIndex = outputFile.getAbsolutePath.lastIndexOf(".vcf")
+      intermResult = new File(
+        (if (sIndex != -1) outputFile.getAbsolutePath.substring(0, sIndex)
+         else outputFile.getAbsolutePath) + ".all_samples.vcf")
+
+      val combineVariants = gatk.CombineVariants(this, outputPerSample, intermResult)
+      combineVariants.genotypemergeoption = Some("UNIQUIFY")
+      add(combineVariants)
+    }
+
+    val selectVariants = new SelectVariants(this)
+    selectVariants.variant = intermResult
+    selectVariants.sample_name = tumorSamples
+
+    val file: File = new File(config("output_dir").asFile, ".renameSamples.txt")
+    file.deleteOnExit()
+    IoUtils.writeLinesToFile(file, renameSamples)
+
+    add(selectVariants | BcftoolsReheader(this, file) | new Bgzip(this) > outputFile)
+
+    add(IGVToolsIndex(this, intermResult))
+    add(Tabix(this, outputFile))
+
+  }
+
+  def addJobForPair(pair: TumorNormalPair, outFile: File): Unit = {
+    val muTect2 =
+      gatk.MuTect2(this, inputBams(pair.tumorSample), inputBams(pair.normalSample), outFile)
+    // TODO add also BQSR file?
```

Comment:
If I understand correct, then running first GATK's PrintReads with the bqsr-file, and then MuTect2 with the modified bam, will give the same result. I think it would be better to stick to that option. The reason is that MuTect2 expects one bqsr-file and that means that BaseRecalibrator should be run with 2 bams (found this from their forum, link below). And this would mean making changes in Biopet in how BaseRecalibrator is called, and I'd prefer not to change that. Cause what is needed is actually there already - PrintReads can be used.

http://gatkforums.broadinstitute.org/gatk/discussion/8036/mutect2-with-bqsr-flag